src/heap.rs

```diff
@@ -487,7 +492,7 @@ impl HeapGuard {
         unsafe {
             ActiveUnsized::write_heap(ptr, self.aheap);
 
-            Ok(ActiveUnsized::data(ptr))
+            Ok(ActiveUnsized::data(ptr, layout))
```

Comment:
AFAICT, this is the only place where `ActiveUnsized::data` is called. In `alloc_raw`, we just calculated the layout and offset above, and then we call `ActiveUnsized::data` and pass in the layout, and that function then calculates the layout and offset a second time. We could avoid a second redundant layout calculation if we saved the offset returned by the first call to `ActiveUnsized::layout`, and changed the `data` function to just take the offset instead of the layout.

Not sure if that's super important or not, just a thought.